poly/src/domain/mod.rs

```diff
@@ -46,6 +46,17 @@ pub trait EvaluationDomain<F: FftField>:
     /// having `num_coeffs` coefficients.
     fn new(num_coeffs: usize) -> Option<Self>;
 
+    /// Construct a domain
+    fn new_subgroup(subgroup_size: usize) -> Option<Self>;
```

Comment:
Is this method different from `new`? That is, is the output checked to be a multiplicative subgroup of `self`? If so, we should make this consistent with `get_coset`, and name the method `get_subgroup`

Reply:
Not sure I fully understand. 

My understanding is that new(num_coefficients) actually determines a subgroup size and new_subgroup(subgroup_size) is provided with a subgroup size.

Reply:
It's effectively the same; the `new_subgroup` seems to assume that its input is a power of 2, whereas `new` does the rounding. Since the two are basically the same, I'd prefer to avoid duplicating the API, and just keep `new` which rounds up.